### PR TITLE
Made resample_sox more flexible

### PR DESCRIPTION
**Pull request by boeddeker**

### CODE REVIEW

File: paderbox/transform/module_resample.py [hunk 1]
```diff
@@ -50,11 +50,14 @@ def resample_sox(signal: np.ndarray, *, in_rate, out_rate):
         signal: Signal as one-dimensional np.ndarray: Shape (T,)
         in_rate: Probably as an integer
         out_rate: Probably as an integer
+        normalize: If True, normalize input to be between -1 and 1, befor
+            calling sox. After resampling, the normalization is reverted.
+            Sox has the issue, to clip some values, if they are above 1.
```
Comment (thequilo) [inline paderbox/transform/module_resample.py:55]: This paragraph has a few typos, maybe write:

If True, the input is normalized to the range between -1 and 1 before calling sox. The normalization is reverted after resampling. Sox clips some values if they exceed 1.

File: paderbox/transform/module_resample.py [hunk 2]
```diff
@@ -50,11 +50,14 @@ def resample_sox(signal: np.ndarray, *, in_rate, out_rate):
...
 
     Returns: Resampled version with same dtype as input.
 
     """
-    assert signal.dtype == np.float32, (
+    assert signal.dtype in [np.float32, np.float64], (
```
Comment (thequilo) [inline paderbox/transform/module_resample.py:60]: The error message sounds strange to me and has float32 hard-coded. Maybe write "The call to SOX just supports float32 and float64, but signal.dtype={signal.dtype}."

File: paderbox/transform/module_resample.py [hunk 3]
```diff
@@ -89,24 +92,30 @@ def resample_sox(signal: np.ndarray, *, in_rate, out_rate):
...
     )
 
-    # This rescaling is necessary since SOX introduces clipping, when the
-    # input signal is much too large.
-    # We normalize each channel independently to avoid rounding errors leading
-    # to the channel doc test above to fail randomly.
-    normalizer = 0.95 / np.max(np.abs(signal), keepdims=True, axis=-1)
-    signal = normalizer * signal
+    if normalize:
+        # This rescaling is necessary since SOX introduces clipping, when the
```
Comment (thequilo) [inline paderbox/transform/module_resample.py:96]: Remove comma before when

File: paderbox/transform/module_resample.py (hunk 1 above)
Comment (boeddeker) [reply paderbox/transform/module_resample.py]: Done

File: paderbox/transform/module_resample.py (hunk 2 above)
Comment (boeddeker) [reply paderbox/transform/module_resample.py:60]: Done

File: paderbox/transform/module_resample.py (hunk 3 above)
Comment (boeddeker) [reply paderbox/transform/module_resample.py]: Done

**Review by thequilo — APPROVED** [review]

**Pull request merged**